cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -861,9 +861,9 @@ def configure_cdk_addons():
 
     set_state('cdk-addons.configured')
     if ks:
-        set_state('keystone.cdk-addons.configured')
+        leader_set({'keystone-cdk-addons-configured': True})
     else:
-        remove_state('keystone.cdk-addons.configured')
+        leader_set({'keystone-cdk-addons-configured': False})
```

Comment:
If you want this to clear the `leadership.set.keystone-cdk-addons-configured` flag, then I think you'll need to set this to `None` instead of `False`. See https://github.com/stub42/layer-leadership/blob/899242c9e3adb3f682ec9db9bc0acdb4b083cfbb/reactive/leadership.py#L54-L55

Reply:
The only case where I see a problem here is if the user relates to keystone, then removes the relation. Since `keystone.credentials.configured` and `leadership.set.keystone-cdk-addons-configured` never get cleared, but `keystone.apiserver.configured` does, the `keystone_kick_apiserver` handler will run on every hook invocation.

Reply:
are you requesting a @when_not handler to clear flags or just change the false set to None?